src/absil/il.fs

```diff
@@ -18,6 +18,7 @@ open System.Collections
  
 let logging = false 
 
+#if FX_RUNNING_ON_MONO
```

Comment:
Yes it would be simpler to wrap the unsupported stuff in a normal #ifdef and leave the majority of the `runningOnMono` stuff unchanged.  Barring that, define something like `NO_MONO_SUPPORT` and remove incompatible bits in a targeted way, again leaving as much of the `runningOnMono` stuff in place as possible - then the current desktop projects don't need a new RUNNING_ON_MONO define. The name `RUNNING_ON_MONO` is misleading in teh first place as this is a compile-time directive, not runtime.